### ISSUE TITLE
Trouble with Passbolt mail config

### ISSUE THREAD

**Issue by Jpeters1990** [passbolt/passbolt_api#119]
Hey guys,

I've run into some trouble sending emails and i hope you guys can help me out.
So, the issue is that cake / passbolt doesnt seem to send emails and fails silently.
I checked every error.log i could find. The one in app, the one in tmp/logs/, even the apache one. No errors to be found.

I have followed this issue: [#84](https://github.com/passbolt/passbolt_api/issues/84) because it feels like the same issue.

My config:
```
   public $default = array(
        'transport' => 'Smtp',
        'from' => array('passbolt@domain.nl' => 'Passbolt'),
        'host' => 'mail.domain.nl',
        'port' => 587,
        'timeout' => 30,
        'username' => 'passbolt@domain.nl',
        'password' => '***********',
        'tls'=> true,
    );
```

I tried both tls true and false in reaction to issue 84, same output (nothing).

Executing EmailQueue.sender like:
`sudo -u {user} ./app/Console/cake EmailQueue.sender > ./app/logs/mail.log`
where app/logs/ is a directory created by me. Still no output.

I've checked the .preview, the emails are there like expected.

Once i got an error in the commandline
`Warning Error: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed in [/home/passbolttheburo/public_html/lib/Cake/Network/CakeSocket.php, line 451]

SMTP server did not accept the connection or trying to connect to non TLS SMTP server using TLS.`
This was only once, so im not really sure if this is related. It might have occured while i was trying something on my email domain.

Can you tell me the next step in debugging further? Do i need more information?

Tnx in advance,
Jeroen

**Comment by cedricalfonsi** [comment]
Hello @Jpeters1990,

I tried many scenarios but I'm unable to reproduce your issue.
Whatever I'm trying I got logs :
- Wrong host : Cannot assign requested address
- Wrong username : SMTP server did not accept the username
- Wrong password : SMTP server did not accept the password
- Wrong port : Connection timed out
- Success : Email EMAIL_HASH was sent

Could you please check :
- Emails are not in your spam
- The {user} you used to execute the script is well your http user
- The http user has the right to write in ./app/logs/mail.log

Ensure you have emails to send. You can create a password before executing the command :
`sudo -u {user} ./app/Console/cake EmailQueue.sender`

Hope that it will help you

**Comment by Jpeters1990** [comment]
Hey @cedricalfonsi,

Thanks for your quick reply!

I checked:
Emails are not in my spam folder.
The user was indeed the same as my http user
And he has the right to write the logs.

But you set me on the right path. Just trying to break my own config to generate some errors i stumbled again on the console error about the SSL certificate (it was still not logging any other errors in the log files). I tried port 25 without TLS in the config and that worked.. it started sending emails.

As to why it is not logging, i really don't know. I will try and figure it out by reviewing all permissions this week. For now we can close this issue because i clearly went wrong somewhere.

I want to thank you for helping!

Jeroen

**Comment by stripthis** [comment]
Glad we could help @Jpeters1990. Feel free to do a final debrief on the issue if you manage to make it work. That may help users with similar problems in the future.

**Comment by glamic-ep** [comment]
I think it would be helpful if the `EmailQueue.sender` would emit something like "Nothing to do." if there are no messages to process. That would maybe save some confusion.

For me the issue with mails was that I had no credentials to configure so I left the default empty string (''). But this was wrong. It should have been `null` instead.

Also I found that it is maybe better to put `/bin/sh` or `/bin/bash` in front of the command in crontab. Also the user has to be able to write the logfile.

**Comment by stripthis** [comment]
@glamic-ep I believe the absence of output when there is nothing to do is to avoid clutering the logs. You can create a separate issue to request an optional parameter like --verbose to add to add the behavior you are describing.
